plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerContainerNameGenerator.java

```diff
@@ -0,0 +1,133 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine;
+
+import org.eclipse.che.api.core.model.machine.Machine;
+import org.eclipse.che.api.core.model.machine.MachineConfig;
+import org.eclipse.che.commons.user.User;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class used for generation docker container name or parse important information from docker container name.
+ *
+ * @author Alexander Andrienko
+ */
+public class DockerContainerNameGenerator {
+    private static final String WORKSPACE_ID_REGEX      = "workspace[\\da-z]{16}";
+    private static final String MACHINE_ID_REGEX        = "machine[\\da-z]{16}";
+
+    private static final String  CONTAINER_NAME_REGEX   = "^" + WORKSPACE_ID_REGEX + '_' + MACHINE_ID_REGEX + "_[a-z0-9_-]+_[a-z0-9_-]+$";
```

Comment:
Yes, @sleshchenko you're right.
But I think we don't need such regexp at all because we won't use that part, only WSID, MachineID. So @AndrienkoAleksandr we can replace it with `_.+`